src/features/user-profile/pages/UserProfilePage.tsx

```diff
@@ -297,20 +355,38 @@ export const UserProfilePage: FC = () => {
                       </p>
 
                       <div className='d-flex align-items-center justify-content-center flex-column'>
-                        <div className='mt-3'>
-                          <UserProfilePicture user={user} size='md' radius={128} />
-                        </div>
-
-                        <div className='w-100 d-grid gap-2'>
-                          <LoadingButton
-                            loading={isLoadingUpdateProfilePicture}
+                        <div className='mt-3 mb-3'>
+                          <UserProfilePictureContainer
+                            hover={isMouseOverProfilePicture}
                             onClick={() => inputFile.current.click()}
-                            variant='default'
-                            className='mt-3'
+                            onMouseEnter={() => setIsMouseOverProfilePicture(true)}
+                            onMouseLeave={() => setIsMouseOverProfilePicture(false)}
```

Comment:
We should use `:hover` in css instead of all this extra javascript and state here. This entire hover interaction can be done in css, only the click on the profile picture should be in javascript.

```css
&:hover::after {
  /* draw the overlay */
}

.the-camera-icon {
  display: none;
  visibility: hidden;
}
&:hover {
  cursor: pointer;
  .the-camera-icon {
    display: block;
    visibility: visible;
  }
}
```

Reply:
Are you saying that we shouldn't make use of the DimmableContent component? That's why I implemented it in this way. Instead of having multiple pieces of overlay logic, I thought it would make sense to expand what that component can do and make use of it here.

Reply:
Thank you for the code. I think I'll be able to manage. I want to see if I can still use the DimmableContent component while staying in line with what you are saying.